GVFS/GVFS/CommandLine/DehydrateVerb.cs

```diff
@@ -145,31 +164,139 @@ protected override void Execute(GVFSEnlistment enlistment)
                 // Local cache and objects paths are required for TryDownloadGitObjects
                 this.InitializeLocalCacheAndObjectsPaths(tracer, enlistment, retryConfig, serverGVFSConfig: null, cacheServer: null);
 
-                if (this.TryBackupFiles(tracer, enlistment, backupRoot))
+                if (string.IsNullOrEmpty(this.Folders))
                 {
-                    if (this.TryDownloadGitObjects(tracer, enlistment, retryConfig) &&
-                        this.TryRecreateIndex(tracer, enlistment))
+                    this.RunFullDehydrate(tracer, enlistment, backupRoot, retryConfig);
+                }
+                else
+                {
+                    string[] folders = this.Folders.Split(new[] { ";" }, StringSplitOptions.RemoveEmptyEntries);
+
+                    if (folders.Length > 0)
                     {
-                        // Converting the src folder to partial must be the final step before mount
-                        this.PrepareSrcFolder(tracer, enlistment);
-                        this.Mount(tracer);
+                        if (cleanStatus)
+                        {
+                            this.DehydrateFolders(tracer, enlistment, backupRoot, folders);
+                        }
+                        else
+                        {
+                            this.ReportErrorAndExit("Must have a clean git status to dehydrate folders.");
+                        }
+                    }
+                    else
+                    {
+                        this.ReportErrorAndExit("No folders to dehydrate.");
+                    }
+                }
+            }
+        }
+
+        private void DehydrateFolders(JsonTracer tracer, GVFSEnlistment enlistment, string backupRoot, string[] folders)
+        {
+            List<string> foldersToDehydrate = new List<string>();
+
+            if (!this.ShowStatusWhileRunning(
+                () =>
+                {
+                    string ioError;
+                    foreach (string folder in folders)
+                    {
+                        string fullPath = Path.Combine(enlistment.WorkingDirectoryRoot, folder);
+                        string backupPath = Path.Combine(backupRoot, folder);
+                        if (this.fileSystem.DirectoryExists(fullPath))

```

Comment:
What were your thoughts here?  Do we need to treat them differently?  They will be in the backup location but we don't do anything with that unless maybe manually moving things back into the repo?

Reply:
Okay I did some testing and added functional tests but here are a couple of points.

>I think the scenario I was wondering about is a bit different. What would/should happen in a scenario like this:
```
// Delete the projected directory "Bar".  A tombstone will be created for "Bar"
C:\Repos\Foo\src>rmdir /s/q Bar

// Dehydrate Bar, what should happen?
C:\Repos\Foo\src>gvfs dehydrate Bar
```

In this case the dehydrate would fail because there would not be a clean git status but you could commit and then dehydrate which is what your second example does but what I have found is that folder tombstones appear as junctions when unmounted.  This is the case whether or not there is another mount running.  So it doesn't get `{folder} did not exist to dehydrate.` but is able to delete the junction.  So I think we are good.